crates/sol-types/src/coder/decoder.rs

```diff
@@ -31,14 +31,35 @@ pub struct Decoder<'de> {
 
 impl fmt::Debug for Decoder<'_> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut body = self
+            .buf
+            .chunks(32)
+            .map(hex::encode_prefixed)
+            .collect::<Vec<_>>();
+        body[self.offset / 32].push_str(" <-- Next Word");
+
         f.debug_struct("Decoder")
-            .field("buf", &hex::encode_prefixed(self.buf))
+            .field("buf", &body)
             .field("offset", &self.offset)
             .field("validate", &self.validate)
             .finish()
     }
 }
 
+impl fmt::Display for Decoder<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut body = self
+            .buf
+            .chunks(32)
+            .enumerate()
+            .map(|(i, chunk)| format!("0x{:0>4x}: {}", i * 32, hex::encode_prefixed(chunk)))
+            .collect::<Vec<_>>();
```

Comment:
we can write this in a for loop with `writeln!` instead of allocating if we cared